stock_available_to_promise_release/models/stock_move.py

```diff
@@ -704,3 +704,29 @@ def _get_new_picking_values(self):
         values = super()._get_new_picking_values()
         values["release_policy"] = values["move_type"]
         return values
+
+    def write(self, vals):
+        if self.env.context.get("in_merge_mode") and "product_uom_qty" in vals:
+            # when a move is merged, we need to unrelease it if the quantity
+            # is changed
+            new_qty = vals["product_uom_qty"]
+            to_unrelease = self.filtered(
+                lambda m: float_compare(
+                    m.product_uom_qty,
+                    new_qty,
+                    precision_rounding=m.product_uom.rounding,
+                )
+                != 0
+            )
+            to_unrelease.unrelease()
```

Comment:
You should not unrelease if it is not allowed (like started)
```suggestion
            to_unrelease.unrelease(safe_unrelease=True)
```